components/script/dom/document.rs

```diff
@@ -1975,6 +1978,16 @@ impl Document {
                            ReflowQueryType::NoQuery,
                            ReflowReason::ElementStateChanged);
     }
+
+    pub fn serialize(&self) -> Fallible<DOMString> {
```

Comment:
What would you suggest? I put here because it was meant to serialize a Document.